libcloud/storage/drivers/backblaze_b2.py

```diff
@@ -292,7 +316,7 @@ def download_object_as_stream(self, obj, chunk_size=None):
                                 success_status_code=httplib.OK)
 
     def upload_object(self, file_path, container, object_name, extra=None,
-                      verify_hash=True, headers=None):
+                      verify_hash=True, headers=None, iterator=None):
```

Comment:
There is already a method for that - `upload_object_via_stream`. That method takes an iterator, while `upload_object` takes a file path.

The reason I didn't implement `upload_object_via_stream` is because Backblaze API doesn't support streaming / multi part uploads. This means we need to buffer the whole file content in memory which is not ideal (especially for large files) and implementing `upload_object_via_stream` doesn't provide much value besides convenience if you already have an iterator object.

Having said that, I'm fine with implementing `upload_object_via_stream` for convenience, we just need to document and make it clear that this method does not perform streaming upload and that the whole file is read and buffered in memory.